Tests/AblyChatTests/DefaultRoomPresenceTests.swift

```diff
@@ -0,0 +1,465 @@
+
+        // Then
+        #expect(defaultPresence.channel.name == "basketball::$chat::$chatMessages")
+    }
+
+    // MARK: CHA-PR3
+
+    // @spec CHA-PR3a
+    // @spec CHA-PR3e
+    @Test
+    func usersMayEnterPresence() async throws {
+        // Given
+        let realtimePresence = MockRealtimePresence(members: ["client1"].map { .init(clientId: $0) })
+        let channel = MockRealtimeChannel(name: "basketball::$chat::$chatMessages", mockPresence: realtimePresence)
+        let featureChannel = MockFeatureChannel(channel: channel, resultOfWaitToBeAblePerformPresenceOperations: .success(()))
+        let defaultPresence = await DefaultPresence(featureChannel: featureChannel, roomID: "basketball", clientID: "client2", logger: TestLogger())
+
+        // When
+        try await defaultPresence.enter(data: ["status": "Online"])
+
+        // Then
+        let presenceMembers = try await defaultPresence.get()
+        #expect(presenceMembers.map(\.clientID).sorted() == ["client1", "client2"])
+        let client2 = presenceMembers.filter { member in
+            member.clientID == "client2" && member.data?.objectValue?["status"]?.stringValue == "Online"
+        }
+        #expect(client2 != nil)
+    }
+
+    // @specOneOf(1/2) CHA-PR3d
+    @Test
+    func usersMayEnterPresenceWhileAttaching() async throws {
+        // Given: A DefaultRoomLifecycleManager, with an ATTACH operation in progress and hence in the ATTACHING status
+        let lifecycleManager = MockRoomLifecycleManager(resultOfWaitToBeAblePerformPresenceOperations: .success(()))
+
+        // Given: A DefaultPresence with DefaultFeatureChannel and MockRoomLifecycleContributor
+        let realtimePresence = MockRealtimePresence(members: ["client1"].map { .init(clientId: $0) })
+        let channel = MockRealtimeChannel(name: "basketball::$chat::$chatMessages", mockPresence: realtimePresence)
+        let contributor = createMockContributor(feature: .presence, attachBehavior: .complete(.success))
+        let featureChannel = DefaultFeatureChannel(channel: channel, contributor: contributor, roomLifecycleManager: lifecycleManager)
+        let defaultPresence = await DefaultPresence(featureChannel: featureChannel, roomID: "basketball", clientID: "mockClientId", logger: TestLogger())
+
+        // When: The room is in the attaching state
+        let roomStatusSubscription = await lifecycleManager.onRoomStatusChange(bufferingPolicy: .unbounded)
+
+        async let _ = lifecycleManager.performAttachOperation()
+
+        // Wait for room to become ATTACHING
+        _ = try #require(await roomStatusSubscription.attachingElements().first { _ in true })
+
+        // When: And presence enter is called
+        try await defaultPresence.enter()
+
+        // Then: The manager was waiting for its room status to change before presence `enter` was called
```

Comment:
There seem to be several long tests for these "wait to be able to perform presence operations" spec points, even though each of them has a single line of implementation; namely `try await featureChannel.waitToBeAbleToPerformPresenceOperations(requestedByFeature: RoomFeature.presence)`. So shouldn't the tests be correspondingly simple (i.e. check that this method was called and check how it behaves when this method succeeds or fails)?